src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingHistoryServiceTest.kt

```diff
@@ -0,0 +1,213 @@
+
+    verify(bookingHistoryRepository).saveAndFlush(historyCaptor.capture())
+
+    with(historyCaptor.firstValue) {
+      historyType isEqualTo HistoryType.CREATE
+      createdTime isCloseTo LocalDateTime.now()
+      hearingType isEqualTo courtBooking.hearingType
+      comments isEqualTo "Court hearing comments"
+      videoUrl isEqualTo courtBooking.videoUrl
+      createdBy isEqualTo CREATED_BY
+      createdTime isCloseTo LocalDateTime.now()
+      appointments() hasSize 1
+      with(appointments().first()) {
+        prisonCode isEqualTo MOORLAND
+        prisonerNumber isEqualTo "A1234AA"
+        appointmentType isEqualTo AppointmentType.VLB_COURT_MAIN.name
+        startTime isEqualTo LocalTime.of(9, 30)
+        endTime isEqualTo LocalTime.of(10, 30)
+        prisonLocKey isEqualTo moorlandLocation.key
+      }
+    }
+  }
+
+  @Test
+  fun `Should create history for a probation booking`() {
+    val probationBooking = probationBooking()
+      .addAppointment(
+        prisonCode = MOORLAND,
+        prisonerNumber = "A1234AA",
+        appointmentType = AppointmentType.VLB_PROBATION.name,
+        date = tomorrow(),
+        startTime = LocalTime.of(9, 30),
+        endTime = LocalTime.of(10, 30),
+        locationKey = moorlandLocation.key,
+      )
+
+    service.createBookingHistoryForProbation(HistoryType.CREATE, probationBooking)
+
+    verify(bookingHistoryRepository).saveAndFlush(historyCaptor.capture())
+
+    with(historyCaptor.firstValue) {
+      historyType isEqualTo HistoryType.CREATE
+      createdTime isCloseTo LocalDateTime.now()
+      probationMeetingType isEqualTo probationBooking.probationMeetingType
+      comments isEqualTo "Probation meeting comments"
+      videoUrl isEqualTo probationBooking.videoUrl
+      createdBy isEqualTo "Probation team user"
+      createdTime isCloseTo LocalDateTime.now()
+      appointments() hasSize 1
+      with(appointments().first()) {
+        prisonCode isEqualTo MOORLAND
+        prisonerNumber isEqualTo "A1234AA"
+        appointmentType isEqualTo AppointmentType.VLB_PROBATION.name
+        startTime isEqualTo LocalTime.of(9, 30)
+        endTime isEqualTo LocalTime.of(10, 30)
+        prisonLocKey isEqualTo moorlandLocation.key
+      }
+    }
+  }
+
+  @Test
+  fun `Should create history for a court booking amendment`() {
+    val courtBooking = courtBooking(CREATED_BY)
+      .addAppointment(
+        prisonCode = MOORLAND,
+        prisonerNumber = "A1234AA",
+        appointmentType = AppointmentType.VLB_COURT_MAIN.name,
+        date = tomorrow(),
+        startTime = LocalTime.of(9, 30),
+        endTime = LocalTime.of(10, 30),
+        locationKey = moorlandLocation.key,
+      )
+
+    service.createBookingHistoryForCourt(HistoryType.AMEND, courtBooking)
+
+    verify(bookingHistoryRepository).saveAndFlush(historyCaptor.capture())
+
+    with(historyCaptor.firstValue) {
+      historyType isEqualTo HistoryType.AMEND
```

Comment:
Perhaps check court attrs populated and probations ones aren't?

Reply:
The CREATE tests check a lot more detail, and its the same function doing the work, so the amend tests only checking that the history type is AMEND.